n3fit/src/n3fit/scripts/n3fit_exec.py

```diff
@@ -145,7 +145,20 @@ def from_yaml(cls, o, *args, **kwargs):
             validation_action = namespace + "validation_pseudodata"
 
             N3FIT_FIXED_CONFIG['actions_'].extend((training_action, validation_action))
-
+        #Theorycovmat flags and defaults
+        N3FIT_FIXED_CONFIG['theory_covmat_flag'] = False
+        if (thconfig:=file_content.get('theorycovmatconfig')) is not None:
+            N3FIT_FIXED_CONFIG['theory_covmat_flag'] = True
+            N3FIT_FIXED_CONFIG['use_thcovmat_in_fitting'] = thconfig.get('use_thcovmat_in_fitting', True) 
+            N3FIT_FIXED_CONFIG['use_thcovmat_in_sampling'] = thconfig.get('use_thcovmat_in_sampling', True)  
+            N3FIT_FIXED_CONFIG['use_user_uncertainties'] = thconfig.get('use_user_uncertainties', False) 
+            N3FIT_FIXED_CONFIG['use_scalevar_uncertainties'] = thconfig.get('use_scalevar_uncertainties', True) 
+        #Sampling flags
+        if (sam_t0:=file_content.get('sampling')) is not None:
+            N3FIT_FIXED_CONFIG['use_t0_sampling'] = sam_t0.get('use_t0', False) 
+            N3FIT_FIXED_CONFIG['separate_multiplicative'] = sam_t0.get('separate_multiplicative', True) 
+        #Fitting flag
+        N3FIT_FIXED_CONFIG['use_t0_fitting'] = file_content.get('fitting').get('use_t0', True)
```

Comment:
Yes but I believe this is necessary to avoid validphys complaining that, for example, theory_covmat_flag does not exist. The moment it exists we do have defaults inside the function but if it does not, then it is a problem.